ethcore/sync/src/chain/handler.rs

```diff
@@ -686,11 +686,11 @@ impl SyncHandler {
 			return Ok(());
 		}
 
-		let item_count = r.item_count()?;
+		let item_count = tx_rlp.item_count()?;
 		trace!(target: "sync", "{:02} -> Transactions ({} entries)", peer_id, item_count);
 		let mut transactions = Vec::with_capacity(item_count);
 		for i in 0 .. item_count {
-			let rlp = r.at(i)?;
+			let rlp = tx_rlp.at(i)?;
 			let tx = rlp.as_raw().to_vec();
```

Comment:
I wanted to get rid of this copy here, but I can't make it work.

Reply:
I don't think it is possible because `Rlp` is essentially a wrapper over `&[u8]`, so one way or another you need to allocate to get `Vec<u8>`.

Reply:
Yeah, my plan was to defer the allocation until we transform the rlp bytes into `UnverifiedTransaction`s. But then the transform to `UnverifiedTransaction` has to happen outside the `ClientIoMessage::Execute` closure. And even then, that closure is an `Fn` and changing that to an `FnOnce` doesn't work because it can't be called when boxed and at that point I gave up.